src/test/java/edu/ucla/library/iiif/fester/verticles/S3BucketVerticleTest.java

```diff
@@ -0,0 +1,189 @@
+
+    private static final String DEFAULT_ACCESS_KEY = "YOUR_ACCESS_KEY";
+
+    private static String s3Bucket = "unconfigured";
+
+    private static AWSCredentials myAWSCredentials;
+
+    @Rule
+    public RunTestOnContext myRunTestOnContextRule = new RunTestOnContext();
+
+    private String myManifestKey;
+
+    /** We can't, as of yet, execute these tests without a non-default S3 configuration */
+    private boolean isExecutable;
+
+    private AmazonS3 myAmazonS3;
+
+    /**
+     * Set up the testing environment.
+     *
+     * @param aContext A test context
+     * @throws Exception If there is trouble starting Vert.x or configuring the tests
+     */
+    @SuppressWarnings("deprecation")
+    @Before
+    public void setUp(final TestContext aContext) throws Exception {
+        final Vertx vertx = myRunTestOnContextRule.vertx();
+        final ConfigRetriever configRetriever = ConfigRetriever.create(vertx);
+        final DeploymentOptions options = new DeploymentOptions();
+        final Async asyncTask = aContext.async();
+
+        configRetriever.getConfig(getConfig -> {
+            if (getConfig.succeeded()) {
+                final JsonObject config = getConfig.result();
+
+                myManifestKey = UUID.randomUUID().toString() + Constants.JSON_EXT;
+
+                // We need to determine if we'll be able to run the S3 integration tests so we can skip if needed
+                if (config.containsKey(Config.S3_ACCESS_KEY) && !config.getString(Config.S3_ACCESS_KEY,
+                        DEFAULT_ACCESS_KEY).equalsIgnoreCase(DEFAULT_ACCESS_KEY)) {
+                    isExecutable = true;
+                }
+
+                vertx.deployVerticle(VERTICLE_NAME, options.setConfig(config), deployment -> {
+                    if (deployment.failed()) {
+                        final Throwable details = deployment.cause();
+                        final String message = details.getMessage();
+
+                        LOGGER.error(details, message);
+                        aContext.fail(message);
+                    }
+
+                    if (myAmazonS3 == null) {
+                        final String s3AccessKey = config.getString(Config.S3_ACCESS_KEY);
+                        final String s3SecretKey = config.getString(Config.S3_SECRET_KEY);
+
+                        // get myAWSCredentials ready
+                        myAWSCredentials = new BasicAWSCredentials(s3AccessKey, s3SecretKey);
+
+                        // instantiate the myAmazonS3 client
+                        myAmazonS3 = new AmazonS3Client(myAWSCredentials);
+                    }
+
+                    s3Bucket = config.getString(Config.S3_BUCKET);
+                    LOGGER.debug(MessageCodes.MFS_067, getClass().getName());
+                    asyncTask.complete();
+                });
+            } else {
+                aContext.fail(getConfig.cause());
+                asyncTask.complete();
```

Comment:
Don't need to complete the asyncTask after an explicit fail. i know this was copied from the bucketeer class but we probably clean it up there too. It generates a warning because the context has already been closed by the time the complete is called.